src/Forum/RecompileFrontendAssets.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Forum;
+
+use Flarum\Foundation\ValidationException;
+use Flarum\Frontend\RecompileFrontendAssets as BaseListener;
+use Flarum\Settings\Event\Saved;
+use Flarum\Settings\Event\Saving;
+use Flarum\Settings\OverrideSettingsRepository;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Events\Dispatcher;
+use Illuminate\Filesystem\FilesystemAdapter;
+use League\Flysystem\Adapter\NullAdapter;
+use League\Flysystem\Filesystem;
+use Less_Exception_Parser;
+
+class RecompileFrontendAssets extends BaseListener
+{
+    public function subscribe(Dispatcher $events)
+    {
+        parent::subscribe($events);
+
+        $events->listen(Saving::class, [$this, 'whenSettingsSaving']);
+    }
+
+    /**
+     * @param Saving $event
+     * @throws ValidationException
+     */
+    public function whenSettingsSaving(Saving $event)
+    {
+        if (isset($event->settings['custom_less'])) {
+            // We haven't saved the settings yet, but we want to trial a full
+            // recompile of the CSS to see if this custom LESS will break
+            // anything. In order to do that, we will temporarily override the
+            // settings repository with the new settings so that the recompile
+            // is effective. We will also use a dummy filesystem so that nothing
+            // is actually written yet.
+
+            $settings = $this->container->make(SettingsRepositoryInterface::class);
+
+            $this->container->extend(
+                SettingsRepositoryInterface::class,
+                function ($settings) use ($event) {
+                    return new OverrideSettingsRepository($settings, $event->settings);
+                }
+            );
```

Comment:
Where is this decorated instance being resolved? Interesting use case for sure. :wink:

Reply:
`Flarum\Forum\Asset\CustomCss`, which is instantiated in a closure within `flarum.forum.assets` defined in `ForumServiceProvider`.

It seems like a *mildly* hacky solution... now that I think of it, perhaps a contextual binding would be better?

Reply:
Decided against the contextual binding; it would imply specific knowledge of the asset compilation system internals, whereas a normal binding is more of a blanket test.